tools/congestion-model/src/model/mod.rs

```diff
@@ -63,13 +84,22 @@ impl Model {
             producer,
             round: 0,
             queues,
+            stats_writer,
+            start_time,
         }
     }
 
     /// execute one round of the model
     pub fn step(&mut self) {
         self.round += 1;
 
+        if let Some(stats_writer) = &mut self.stats_writer {
+            let time = self.start_time + Duration::from_secs(self.round);
+
+            stats_writer.write_field(format!("{:?}", time)).unwrap();
+            stats_writer.write_field(format!("{}", self.round)).unwrap();
+        }
+
```

Comment:
Ideally the `step` function does not need to know about the stats writer at all.
The code running the model (in main.rs) can call a evaluation function after every call to `step()`.
Then we can add all kind of fancy output formatting options in `main.rs` that don't affect how the model runs.
At least that's what I had in mind for the code base architecture.